workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/androidx/WorkflowLifecycleOwner.kt

```diff
@@ -190,32 +191,38 @@ internal class RealWorkflowLifecycleOwner(
       oldLifecycle?.removeObserver(this)
       parentLifecycle?.addObserver(this)
     }
-    updateLifecycle()
+
+    // if (parentLifecycle?.currentState == INITIALIZED) {
+    //  // updateLifecycle(Event.ON_RESUME.targetState)
+    // } else {
+    //
+    // }
+    updateLifecycle(Event.ON_RESUME.targetState)
   }
 
   override fun onViewDetachedFromWindow(v: View) {
     viewIsAttachedToWindow = false
-    updateLifecycle()
+    updateLifecycle(Event.ON_PAUSE.targetState)
   }
 
   /** Called when the [parentLifecycle] changes state. */
   override fun onStateChanged(
     source: LifecycleOwner,
     event: Event
   ) {
-    updateLifecycle()
+    updateLifecycle(event.targetState)
   }
 
   override fun destroyOnDetach() {
     if (!destroyOnDetach) {
       destroyOnDetach = true
-      updateLifecycle()
+      updateLifecycle(Event.ON_DESTROY.targetState)
```

Comment:
Just to make sure the comment doesn't get lost across updates: we can't ship this way. 

- [ ] `destroyOnDetach()` has to mean "move to destroyed the next time `onViewDetachedFromWindow()` is called.